src/NuGet.Core/NuGet.Packaging/Core/comparers/PackageIdentityComparer.cs

```diff
@@ -48,6 +48,22 @@ public PackageIdentityComparer(IVersionComparer versionComparer)
         /// </summary>
         public static PackageIdentityComparer Default { get; } = new PackageIdentityComparer();
 
+        internal static PackageIdentityComparer Version { get; } = new PackageIdentityComparer(VersionComparison.Version);
+        internal static PackageIdentityComparer VersionRelease { get; } = new PackageIdentityComparer(VersionComparison.VersionRelease);
+        internal static PackageIdentityComparer VersionReleaseMetadata { get; } = new PackageIdentityComparer(VersionComparison.VersionReleaseMetadata);
+
+        internal static PackageIdentityComparer Get(VersionComparison versionComparison)
+        {
+            return versionComparison switch
+            {
+                VersionComparison.Default => Default,
+                VersionComparison.Version => new PackageIdentityComparer(VersionComparison.Version),
+                VersionComparison.VersionRelease => new PackageIdentityComparer(VersionComparison.VersionRelease),
+                VersionComparison.VersionReleaseMetadata => new PackageIdentityComparer(VersionComparison.VersionReleaseMetadata),
```

Comment:
These are intended to return the static instances allocated above, right?

```suggestion
                VersionComparison.Version => Version,
                VersionComparison.VersionRelease => VersionRelease,
                VersionComparison.VersionReleaseMetadata => VersionReleaseMetadata,
```

Reply:
Yes, my bad. Thanks for spotting.